contracts/SwapKiwi.sol

```diff
@@ -200,43 +244,45 @@ contract SwapKiwi is Ownable, IERC721Receiver {
   }
 
   /**
-    * @dev Returns NFTs from SwapKiwi to swap initator.
-    *      Callable only if second user hasn't yet added NFTs.
-    *
-    * @param swapId ID of the swap that the swap participants want to cancel
-    */
-  function cancelSwap(uint256 swapId) external {
+  * @dev Returns NFTs from SwapKiwi to swap initator.
+  *      Callable only if second user hasn't yet added NFTs.
+  *
+  * @param swapId ID of the swap that the swap participants want to cancel
+  */
+  function cancelSwap(uint64 swapId) external {
     require(
       _swaps[swapId].initiator == msg.sender || _swaps[swapId].secondUser == msg.sender,
       "SwapKiwi: Can't cancel swap, must be swap participant"
     );
-      // return initiator NFTs
-      safeMultipleTransfersFrom(
-        address(this),
-        _swaps[swapId].initiator,
-        _swaps[swapId].initiatorNftAddresses,
-        _swaps[swapId].initiatorNftIds
-      );
+    // return initiator NFTs
+    safeMultipleTransfersFrom(
+      address(this),
+      _swaps[swapId].initiator,
+      _swaps[swapId].initiatorNftAddresses,
+      _swaps[swapId].initiatorNftIds,
+      _swaps[swapId].initiatorNftAmounts
+    );
 
     if(_swaps[swapId].secondUserNftAddresses.length != 0) {
       // return second user NFTs
       safeMultipleTransfersFrom(
         address(this),
         _swaps[swapId].secondUser,
         _swaps[swapId].secondUserNftAddresses,
-        _swaps[swapId].secondUserNftIds
+        _swaps[swapId].secondUserNftIds,
+        _swaps[swapId].secondUserNftAmounts
       );
     }
 
     if (_swaps[swapId].initiatorEtherValue != 0) {
       _etherLocked -= _swaps[swapId].initiatorEtherValue;
-      uint amountToTransfer = _swaps[swapId].initiatorEtherValue;
+      uint128 amountToTransfer = _swaps[swapId].initiatorEtherValue;
       _swaps[swapId].initiatorEtherValue = 0;
       _swaps[swapId].initiator.transfer(amountToTransfer);
```

Comment:
Due to possible future gas cost changes, it is discouraged to use address.transfer, instead should use address.call with success verification.